### PR TITLE
docs: modify the make install in how to build Apache APISIX doc

### PR DESCRIPTION
**Pull request by bin-ya**
### Description

Fixes: #6559

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/en/latest/how-to-build.md [hunk 1]
```diff
@@ -125,7 +125,7 @@ Follow the steps below to install Apache APISIX via the source release package.
   cd apisix-${APISIX_VERSION}
   # Create dependencies
   make deps
-  # Install apisix command
+  # Install apisix command (only for luarocks)
   make install
```
Comment (tzssangglass) [inline docs/en/latest/how-to-build.md:129]: rm `make install` is better?

Comment (tzssangglass) [reply docs/en/latest/how-to-build.md:129]: cc @spacewander

Comment (spacewander) [reply docs/en/latest/how-to-build.md:129]: Yes. We can rename this task and claim that the developer should not try to "install" the source.

Comment (tzssangglass) [reply docs/en/latest/how-to-build.md:129]: @bin-ya I think `Install apisix command (only for luarocks)` should be changed to `Install apisix command (developer do not need to follow this step)`

Comment (bin-ya) [reply docs/en/latest/how-to-build.md:129]: ok

**Review by tzssangglass — DISMISSED** [review]

File: docs/en/latest/how-to-build.md [hunk 2]
```diff
@@ -125,7 +125,7 @@ Follow the steps below to install Apache APISIX via the source release package.
   cd apisix-${APISIX_VERSION}
   # Create dependencies
   make deps
-  # Install apisix command
+  # Install apisix command (developer do not need to follow this step)
```
Comment (spacewander) [inline docs/en/latest/how-to-build.md:128]: I am confused. If it is no need, why not remove it?

Comment (tzssangglass) [reply docs/en/latest/how-to-build.md:128]: let's remove it. cc @bin-ya

**Review by tzssangglass — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]

File: docs/en/latest/how-to-build.md [hunk 3]
```diff
@@ -125,8 +125,6 @@ Follow the steps below to install Apache APISIX via the source release package.
   cd apisix-${APISIX_VERSION}
   # Create dependencies
   make deps
-  # Install apisix command
-  make install
```
Comment (spacewander) [inline docs/en/latest/how-to-build.md:129]: We can also remove `make uninstall`?

Comment (bin-ya) [reply docs/en/latest/how-to-build.md:129]: Sorry for my mistake, I'll change it immediately

**Review by spacewander — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]
LGTM

**Pull request merged**